python/triton/config.py

```diff
@@ -0,0 +1,310 @@
+        ...
+
+
+# This is of the form [attr_name, attr_val]
+# TODO: Use tuple instead of list for better typing.
+KernelAttr = list[str | int]
+
+
+class JitHookCompileInfo(TypedDict):
+    key: str
+    signature: dict[KernelParam, str]
+    device: int
+    constants: None
+    num_warps: int
+    num_ctas: int
+    num_stages: int
+    enable_fp_fusion: bool
+    launch_cooperative_grid: bool
+    extern_libs: tuple[tuple[str, str], ...]
+    configs: list[dict[tuple[int, ...], list[KernelAttr]]]
+    specialization_data: str
+    is_warmup: bool
+
+
+class JitHook(Protocol):
+
+    def __call__(self, *, key: str, repr: str, fn: JitFunctionInfo, compile: JitHookCompileInfo, is_manual_warmup: bool,
+                 already_compiled: bool) -> bool | None:
+        ...
+
+
+class runtime:
+    interpret: env_bool = env_bool("TRITON_INTERPRET")
+    debug: env_bool = env_bool("TRITON_DEBUG")
+    override_arch: env_str = env_str("TRITON_OVERRIDE_ARCH")
+
+    launch_enter_hook: LaunchHook | None = None
+    launch_exit_hook: LaunchHook | None = None
+
+    # Hook for inspecting compiled functions and modules
+    jit_cache_hook: JitHook | None = None
+    # Hook to signal that a kernel is done compiling and inspect compiled function.
+    # jit_cache_hook will always be called before compilation and jit_post_compile_hook after.
+    jit_post_compile_hook: JitHook | None = None
+
+
+class language:
+    fp32_default: env_str = env_str("TRITON_F32_DEFAULT")
+    default_fp_fusion: env_bool = env_bool("TRITON_DEFAULT_FP_FUSION", True)
+
+
+class nvidia:
+    cuobjdump: NvidiaTool = get_nvidia_tool("cuobjdump")  #field(default_factory=lambda: _get_nvidia_tool("cuobjdump"))
+    nvdisasm: NvidiaTool = get_nvidia_tool("nvdisasm")  #field(default_factory=lambda: _get_nvidia_tool("nvdisasm"))
+    ptxas: NvidiaTool = get_nvidia_tool("ptxas")  #field(default_factory=lambda: _get_nvidia_tool("ptxas"))
+
+    dump_nvptx: env_bool = env_bool("NVPTX_ENABLE_DUMP")
+    disable_ptxas_opt: env_bool = env_bool("DISABLE_PTXAS_OPT")
+    mock_ptx_version: env_str = env_str("TRITON_MOCK_PTX_VERSION")
+
+    libdevice_path: env_str = env_str("TRITON_LIBDEVICE_PATH")
+    libcuda_path: env_str = env_str("TRITON_LIBCUDA_PATH")
+
+
+class amd:
+    use_buffer_ops: env_bool = env_bool("AMDGCN_USE_BUFFER_OPS")
+    dump_amdgcn: env_bool = env_bool("AMDGCN_ENABLE_DUMP")
+    libhip_path: env_str = env_str("TRITON_LIBHIP_PATH")
+    lld_path: env_str = env_str("TRITON_HIP_LLD_PATH")
+
+    # We use strs so that we can have a default value based on other runtime info
+    use_block_pingpong: env_str = env_str("TRITON_HIP_USE_BLOCK_PINGPONG")
+    use_in_thread_transpose: env_str = env_str("TRITON_HIP_USE_IN_THREAD_TRANSPOSE")
+
+    global_prefetch: env_int = env_int("TRITON_HIP_GLOBAL_PREFETCH")
+    local_prefetch: env_int = env_int("TRITON_HIP_GLOBAL_PREFETCH")
+    use_async_copy: env_bool = env_bool("TRITON_HIP_GLOBAL_PREFETCH")
```

Comment:
This seems consistent with everything I've seen on the AMD side, but I would broadcast this to the AMD team more broadly so that in development work is updated. I can think of 2-3 PRs which are likely gated behind env variables.

Reply:
Sure, seems reasonable. FWIW I tried my best to be backwards compatible with an old workflows